packages/clay-card/src/index.tsx

```diff
@@ -0,0 +1,132 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+import Context from './Context';
+import {CardDisplayType} from './types';
+
+import {AspectRatio, CardAspectRatioType} from './AspectRatio';
+import {Body, CardBodyProps, CardBodyType} from './Body';
+import {CardDescriptionType, Description} from './Description';
+import {CardDetailType, Detail} from './Detail';
+import {CardGroupType, Group} from './Group';
+import {CardHeaderType, Header} from './Header';
+
+export interface CardProps {
+	displayType?: CardDisplayType;
+
+	/**
+	 * Flag that indicates if the card can be selectable.
+	 */
+	selectable?: boolean;
+}
+
+interface Props
+	extends CardProps,
+		React.BaseHTMLAttributes<HTMLAnchorElement | HTMLDivElement> {
+	children:
+		| React.ReactElement<CardBodyProps>[]
+		| React.ReactElement<CardBodyProps>;
+}
+
+const ClayCard: React.FunctionComponent<Props> & {
+	AspectRatio: CardAspectRatioType;
+	Body: CardBodyType;
+	Detail: CardDetailType;
+	Description: CardDescriptionType;
+	Group: CardGroupType;
+	Header: CardHeaderType;
+} = ({
+	children,
+	className,
+	displayType = 'directory',
+	selectable = false,
+	...otherProps
+}) => {
+	const Content: React.FunctionComponent<Props> = ({children}) => (
+		<Context.Provider value={{displayType}}>
```

Comment:
I'm a little hesitant to use context for this use case here, especially because displayType can be `'horizontal interactive'` which seems really flakey to me since if someone passed in `'interactive horizontal '` it would break. Is this card too complex to simply rely on props? Context is nice as times, but can often over-complicate things.

Reply:
"Splitted" into 2 props, horizontal and interactive. What do you think?

Reply:
https://github.com/liferay/clay/pull/2007/commits/81f11acaea996f1e5e59fa07ffa2b508beb2d439

Reply:
Solved